apps/desktop-wallet/src/storage/prices/pricesActions.ts

```diff
@@ -0,0 +1,102 @@
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { CHART_DATE_FORMAT, TOKENS_QUERY_LIMIT } from '@alephium/shared'
+import { createAsyncThunk } from '@reduxjs/toolkit'
+import dayjs from 'dayjs'
+import { chunk } from 'lodash'
+
+import client from '@/api/client'
+import { TokenHistoricalPrice, TokenPriceEntity, TokenPriceHistoryEntity } from '@/types/price'
+import { isFulfilled } from '@/utils/misc'
+
+export const syncTokenCurrentPrices = createAsyncThunk(
+  'assets/syncTokenCurrentPrices',
+  async ({ verifiedFungibleTokenSymbols, currency }: { verifiedFungibleTokenSymbols: string[]; currency: string }) => {
+    let tokenPrices = [] as TokenPriceEntity[]
+
+    const promiseResults = await Promise.allSettled(
+      chunk(verifiedFungibleTokenSymbols, TOKENS_QUERY_LIMIT).map((verifiedFungibleTokenSymbolsChunk) =>
+        client.explorer.market
+          .postMarketPrices(
+            {
+              currency: currency.toLowerCase()
+            },
+            verifiedFungibleTokenSymbolsChunk
+          )
+          .then((prices) =>
+            prices.map((price, index) => ({
+              symbol: verifiedFungibleTokenSymbolsChunk[index],
+              price
+            }))
+          )
+      )
+    )
+
+    tokenPrices = promiseResults.filter(isFulfilled).flatMap((r) => r.value)
+
+    return tokenPrices
+  }
+)
+
+export const syncTokenPriceHistories = createAsyncThunk(
+  'assets/syncTokenPriceHistories',
+  async ({ verifiedFungibleTokenSymbols, currency }: { verifiedFungibleTokenSymbols: string[]; currency: string }) => {
+    let tokenPriceHistories = [] as TokenPriceHistoryEntity[]
+
+    const promiseResults = await Promise.allSettled(
+      verifiedFungibleTokenSymbols.map((symbol) =>
+        client.explorer.market
+          .getMarketPricesSymbolCharts(symbol, {
+            currency: currency.toLowerCase()
+          })
+          .then((rawHistory) => {
+            const today = dayjs().format(CHART_DATE_FORMAT)
+            let history = [] as TokenHistoricalPrice[]
+
+            if (rawHistory.timestamps && rawHistory.prices) {
+              const pricesHistoryArray = rawHistory.prices
+
+              history = rawHistory.timestamps.reduce((acc, v, index) => {
+                const itemDate = dayjs(v).format(CHART_DATE_FORMAT)
+                const isDuplicatedItem = !!acc.find(({ date }) => dayjs(date).format(CHART_DATE_FORMAT) === itemDate)
```

Comment:
Are we sure this still happens (duplicated dates)? Asking because this is quite heavy, using `find` for every item in the timestamps list.

Reply:
We'd need to ask Thomas or inspect it ourselves.

Reply:
Yep, it's still needed

<img width="468" alt="image" src="https://github.com/alephium/alephium-frontend/assets/1579899/37e67523-a902-4dd7-8f88-9c9d451747a9">